mos-platform/neo6502/api/api-internal.h

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+#include <stdint.h>
+#include <string.h>
+
+#ifndef _NEO_API_INTERNAL_H
+#define _NEO_API_INTERNAL_H
+
+/**
+ * @brief Given an input C-format string FROM, provide a Pascal-format string FROM_p using VLAs.
+ */
+#define PASCALIZE_INPUT(FROM) \
```

Comment:
It's considered good macro practice (coming from the LISP world) to have macros that bind new public names do so by taking the name as an argument. Even though it can always be advertised as a formula derived from the given name, it requires less knowledge of the macro to read `PASCALIZE_INPUT(str_p, str); oh that must have defined this reference to str_p;` than `PASCALIZE_INPUT(str); str_p appears outta nowhere`.